packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -111,8 +119,13 @@ class _PosixUtils extends OperatingSystemUtils {
   _PosixUtils() : super._private();
 
   @override
-  ProcessResult makeExecutable(File file) {
-    return processManager.runSync(<String>['chmod', 'a+x', file.path]);
+  void makeExecutable(File file) {
+    chmod(file, 'a+x');
+  }
+
+  @override
+  void chmod(FileSystemEntity entity, String mode) {
+    processManager.runSync(<String>['chmod', mode, entity.path]);
```

Comment:
This call can throw a `ProcessException`. It can also return a `ProcessResult` with a non-zero exit status. In both of those cases, I think we should probably printError and try to continue.

Reply:
The situation I'd like to avoid is a user with a hosed setup flooding crash reporting with unhandled exceptions. I'm okay with the caller handling the exception. Another option would be for `chmod` to return a `bool` that is false if something went wrong, then the caller can give a better `printError` or `throwToolExit` as appropriate.

Reply:
Ok, I changed it to print a trace statement instead of throwing, which is closer to how `makeExecutable()` was working before this change as well.